### PR TITLE
[DevEx]: components name instance in pascalcase

### LINKED ISSUE #396: [DevEx]: Update component <template> code to use PascalCase for custom component name instances
### Improvement Summary

The vue style guides recommend using PascalCase for custom component instances in template code e.g. `<BaseDialog>` rather than `<base-dialog>.` We should update the entire clientside codebase to reflect this change.

### Detailed Description

This should entail finding all instances of cuttle-specific custom components and pages and referencing them with `<PascalCase>` tags in template code rather than the current `<kepbab-case>` code. In addition we should identify whether it's possible to configure the linter to error on other capitalization strategies and whether we can configure either eslint or vscode to auto-fix capitalization of custom components to transform them into PascalCase.

This issue was kicked off from a comment in #389

### PR DESCRIPTION
**Pull request by haaarsh619**
## Issue number #396 

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #396 

## Please check the following

- [X] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [X] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

### CODE REVIEW

File: src/components/GameView/CounterDialog.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <choose-whether-to-counter-dialog
+  <ChooseWhetherToCounterDialog
```
Comment (seriouslysean) [inline src/components/GameView/CounterDialog.vue:2]: This is crazy long, not for this PR but maybe we can figure out better naming for some of these.

**Review by seriouslysean — COMMENTED** [review]
All the files changed look in line with the designation to me, leaving finally approval to @itsalaidbacklife -- just one callout that isn't a blocker by any means.

File: src/components/GameView/CannotCounterDialog.vue [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-        Your opponent has played the
-        <game-card-name :card-name="oneOff.name" />
+        Your opponent has played the 
+        <GameCardName :card-name="oneOff.name" />
         as a one-off
-        <span v-if="target">
-          targeting your
-          <game-card-name :card-name="target.name" />
+        <span v-if="target"> 
+          targeting your
```
Comment (itsalaidbacklife) [inline src/components/GameView/CannotCounterDialog.vue:14]: Don't think we need the whitespace
```suggestion
        <span v-if="target">
          targeting your
```

File: src/components/GameView/CounterDialog.vue (hunk 1 above)
Comment (itsalaidbacklife) [reply src/components/GameView/CounterDialog.vue:2]: I'm open to suggestions. I tend to favor verbose-and-explicit to concise but any time we can be clear and concise is a win

**Review by itsalaidbacklife — COMMENTED** [review]
Looks good. I have one question about the config in the comments

**Review by itsalaidbacklife — APPROVED** [review]
Nice work. Good linting simplifies maintenance, which is especially valuable during hacktoberfest. Thank you!

**Pull request merged**